core/src/test/java/ai/timefold/solver/core/impl/testdata/domain/shadow/wrong_cascade/TestdataCascadingPiggyback.java

```diff
@@ -4,33 +4,36 @@
 import ai.timefold.solver.core.api.domain.variable.CascadingUpdateShadowVariable;
 import ai.timefold.solver.core.api.domain.variable.InverseRelationShadowVariable;
 import ai.timefold.solver.core.api.domain.variable.NextElementShadowVariable;
+import ai.timefold.solver.core.api.domain.variable.PiggybackShadowVariable;
 import ai.timefold.solver.core.api.domain.variable.PreviousElementShadowVariable;
 import ai.timefold.solver.core.impl.domain.entity.descriptor.EntityDescriptor;
 import ai.timefold.solver.core.impl.domain.solution.descriptor.SolutionDescriptor;
-import ai.timefold.solver.core.impl.testdata.domain.cascade.single_var.shadow_var.TestdataSingleCascadingEntity;
-import ai.timefold.solver.core.impl.testdata.domain.cascade.single_var.shadow_var.TestdataSingleCascadingSolution;
+import ai.timefold.solver.core.impl.testdata.domain.cascade.single_var.TestdataSingleCascadingEntity;
+import ai.timefold.solver.core.impl.testdata.domain.cascade.single_var.TestdataSingleCascadingSolution;
 
 @PlanningEntity
-public class TestdataCascadingNoSources {
+public class TestdataCascadingPiggyback {
 
     public static EntityDescriptor<TestdataSingleCascadingSolution> buildEntityDescriptor() {
         return SolutionDescriptor
                 .buildSolutionDescriptor(TestdataSingleCascadingSolution.class, TestdataSingleCascadingEntity.class,
-                        TestdataCascadingNoSources.class)
-                .findEntityDescriptorOrFail(TestdataCascadingNoSources.class);
+                        TestdataCascadingPiggyback.class)
+                .findEntityDescriptorOrFail(TestdataCascadingPiggyback.class);
     }
 
     @InverseRelationShadowVariable(sourceVariableName = "valueList")
     private TestdataSingleCascadingEntity entity;
     @PreviousElementShadowVariable(sourceVariableName = "valueList")
-    private TestdataCascadingNoSources previous;
+    private TestdataCascadingPiggyback previous;
     @NextElementShadowVariable(sourceVariableName = "valueList")
-    private TestdataCascadingNoSources next;
+    private TestdataCascadingPiggyback next;
     @CascadingUpdateShadowVariable(targetMethodName = "updateCascadeValue")
     private Integer cascadeValue;
+    @PiggybackShadowVariable(shadowVariableName = "cascadeValue")
+    private Integer cascadeValue2;
```

Comment:
This should be no longer possible, right?

Reply:
Yes, there is a new test to cover that. Let me rename the class to make it clear.